### PR TITLE
Added Unit testing with pytest

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
Added initial file structure and sample unit tests. Updated requirements to support unit testing.
Tested with end-to-end testing to ensure no functional changes to LDAR-Sim. Logs will be attached. [unit_testing_logs.zip](https://github.com/LDAR-Sim/LDAR_Sim/files/11626945/unit_testing_logs.zip)

### CODE REVIEW

**Review by SEAJang — CHANGES_REQUESTED** [review]
Could we get some sort of naming / numbering convention for the unit tests, add that into the documentation and change the existing unit tests to follow the convention that is written out? 
Otherwise, it's looks good.

File: LDAR_Sim/testing/unit_testing/test_methods/crew_testing_fixutres.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+from typing import Any
+
```
Comment (KeeganShaw-GIS) [inline LDAR_Sim/testing/unit_testing/test_methods/crew_testing_fixutres.py:2]: File name has a spelling mistake I think?

**Review by KeeganShaw-GIS — CHANGES_REQUESTED** [review]
This looks awesome. Agree with Sally's comments. And just double check the spelling of crew testing

**Comment by ThomasGalesloot** [comment]
> Could we get some sort of naming / numbering convention for the unit tests, add that into the documentation and change the existing unit tests to follow the convention that is written out? Otherwise, it's looks good.

Unit testing has been refactored to have a consistent file structure and naming convention

**Review by KeeganShaw-GIS — APPROVED** [review]

**Pull request merged**